tests/preflight/preflight_test.go

```diff
@@ -0,0 +1,753 @@
+package preflighttest
+
+import (
+	"crypto/rand"
+	"fmt"
+	"io/fs"
+	"io/ioutil"
+	"math/big"
+	"os"
+	"path"
+	"strings"
+	"sync"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/trilioData/tvk-plugins/internal"
+	"github.com/trilioData/tvk-plugins/internal/utils/shell"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+var _ = Describe("Preflight Tests", func() {
+
+	Context("When checks are successful and volume snapshot class value is not provided as input", func() {
+		var (
+			once          sync.Once
+			outputLogData string
+		)
+		BeforeEach(func() {
+			once.Do(func() {
+				var byteData []byte
+				runPreflightChecks(flagsMap)
+				byteData, err = getLogFileData(preflightLogFilePrefix)
+				Expect(err).To(BeNil())
+				outputLogData = string(byteData)
+			})
+		})
+
+		It("Should be able fetch volume snapshot class with its driver matching storage class' provisioner", func() {
```

Comment:
All `IT`s should be around preflight check, should pass or fail. For example - `volume-snapshot-class` should pass, `cluster-access` check should fail, etc.
Create separate validation funcs for each preflight check and put all validation assertions in that func and reuse.